### PR TITLE
fix(cloudformation): Update CKV_AWS_46 to handle base64 encoded userdata

### LINKED ISSUE #4516: CloudFormation EC2Credentials.py does evaluate if the UserData is base64 encoded
**Describe the issue**
The check EC2Credentials.py which checks the UserData of a CloudFormation template against secrets does not work if the userdata are already encoded in base64. 
The check is not returning the correct result and the processing time is very slow.

**Examples**
```yaml
AWSTemplateFormatVersion: "2010-09-09"
Resources:
  Resource0:
    Type: AWS::EC2::Instance
    Properties:
      ImageId: ami-04169656fea786776
      UserData: IyEgL2Jpbi9iYXNoCnN1ZG8gYXB0LWdldCB1cGRhdGUKc3VkbyBhcHQtZ2V0IGluc3RhbGwgLXkgYXBhY2hlMgpzdWRvIHN5c3RlbWN0bCBzdGFydCBhcGFjaGUyCnN1ZG8gc3lzdGVtY3RsIGVuYWJsZSBhcGFjaGUyCmV4cG9ydCBBV1NfQUNDRVNTX0tFWV9JRApleHBvcnQgQVdTX0FDQ0VTU19LRVlfSUQ9QUtJQUlPU0ZPRE5ON0VYQU1QTEUKZXhwb3J0IEFXU19TRUNSRVRfQUNDRVNTX0tFWT13SmFsclhVdG5GRU1JL0s3TURFTkcvYlB4UmZpQ1lFWEFNUExFS0VZCmV4cG9ydCBBV1NfREVGQVVMVF9SRUdJT049dXMtd2VzdC0yCmVjaG8gIjxoMT5EZXBsb3llZCB2aWEgVGVycmFmb3JtPC9oMT4iIHwgc3VkbyB0ZWUgL3Zhci93d3cvaHRtbC9pbmRleC5odG1s

```

**Version (please complete the following information):**
 - Checkov Version [e.g. 22]
 - >2.0.0

**Additional context**
Fix implemented here by trying to decode the base64 string.
https://github.com/j2clerck/checkov/commit/af9abd724520ec21ec5510dfc5db2ef83fd9e6dc

### PR DESCRIPTION
**Pull request by j2clerck**
**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

[//]: # "
    # PR Title
    Be aware that we use the title to create changelog automatically and therefore only allow specific prefixes
    - break:    to indicate a breaking change, this supersedes any of the types
    - feat:     to indicate new features or checks
    - fix:      to indicate a bugfix or handling of edge cases of existing checks
    - docs:     to indicate an update to our documentation
    - chore:    to indicate adjustments to workflow files or dependency updates
    - platform: to indicate a change needed for the platform
    Additionally a scope is needs to be added to the prefix, which indicates the targeted framework, in doubt choose 'general'.
    #    
    Allowed prefixs:
    ansible|argo|arm|azure|bicep|bitbucket|circleci|cloudformation|dockerfile|github|gha|gitlab|helm|kubernetes|kustomize|openapi|sca|secrets|serverless|terraform|general|graph|terraform_plan
    #
    ex.
    feat(terraform): add CKV_AWS_123 to ensure that VPC Endpoint Service is configured for Manual Acceptance
"

## Description

This PR fixes a bug when using checkov would spend minutes to check base64 encoded string for userdata without finding error.

Fixes #4516 

## New/Edited policies (Delete if not relevant)
CKV_AWS_46

### Description
If a UserData is passed as a base64encoded string, the check against secrets will fail to detect one and spend a lot of time, multiple minutes, to perform the scan.

### Fix
The fix is to try to decode the UserData payload to see if it is a base64 encoded string.

## Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] I have added tests that prove my feature, policy, or fix is effective and works
- [X] New and existing tests pass locally with my changes
- [X] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: tests/cloudformation/checks/resource/aws/test_EC2Credentials.py [hunk 1]
```diff
@@ -15,9 +15,10 @@ def test_summary(self):
         test_files_dir = current_dir + "/example_EC2Credentials"
         report = runner.run(root_folder=test_files_dir,runner_filter=RunnerFilter(checks=[check.id]))
         summary = report.get_summary()
+        print(summary)
```
Comment (nimrodkor) [inline tests/cloudformation/checks/resource/aws/test_EC2Credentials.py:18]: @j2clerck No need to log this :)

**Review by nimrodkor — APPROVED** [review]
TY @j2clerck !

File: checkov/cloudformation/checks/resource/aws/EC2Credentials.py [hunk 2]
```diff
@@ -1,10 +1,14 @@
+import base64
+import binascii
 from typing import List
+from base64 import b64decode
```
Comment (gruebel) [inline checkov/cloudformation/checks/resource/aws/EC2Credentials.py:4]: `binascii` is an unused import. wither you can keep the `base64` one and remove `base64 import b64decode` or keep it and remove just the `base64`, but then you need to adjust the usage a few lines lower.

File: checkov/cloudformation/checks/resource/aws/EC2Credentials.py [hunk 3]
```diff
@@ -1,10 +1,14 @@
...
 
 from checkov.common.models.enums import CheckResult, CheckCategories
 from checkov.cloudformation.checks.resource.base_resource_check import BaseResourceCheck
 from checkov.common.util.secrets import get_secrets_from_string
 
 
+
```
Comment (gruebel) [inline checkov/cloudformation/checks/resource/aws/EC2Credentials.py:11]:
```suggestion
```
probably the linter will blame you for adding this extra line 😄

**Review by gruebel — COMMENTED** [review]
great work, just some minor comments

**Pull request merged**